File: 02.Binary Search/binary_search.py
def binary_search(aList, sizeOfList, aVar):
    found = False
    upper = sizeOfList - 1
    lower = 0
    while(not found):
        middle = (upper+lower) // 2
        if aList[middle] == aVar:
            print("{0} at {1} Matched".format(aVar, middle))
            found = True
        elif aList[middle] > aVar:
            upper = middle - 1
        else:
            lower = middle + 1
    
    return found



if __name__ == "__main__":
    integers = [10,20,30,40,50,60,70,80,90,100]
    find = 70

    if binary_search(integers, len(integers), find):
        print("Found")
    else:
        print("Not Found")

File: 02.Binary Search/binary_search.c
#include <stdio.h>

int binary_search(int array[], int size, int var);

int main()
{
    int numbers[] = {10,20,30,40,50,60,70,80,90,100};
    int find = 70;

    if(binary_search(numbers, sizeof(numbers)/4, find)){
        printf("Found\n");
    }
    else{
        printf("Not found\n");
    }

    return 0;
}


int binary_search(int array[], int size, int var){

    int found = 0;
    int middle;
    int upper = size-1;
    int lower = 0;

    while(!found){
        middle = (upper + lower) / 2;
        printf("middle: %d\n", middle);

        if (array[middle] == var) {
            printf("%d at %d Matched\n", var, middle);
            found = 1;
        }

        else if (array[middle] > var) {
            printf("Setting upper\n");
            upper = middle - 1;
        }
        else{
            printf("Setting lower\n");
            lower = middle + 1;
        }
    }

    return found;
}
